modules/ROOT/pages/troubleshoot-adapter-provisioning-issues.adoc

```diff
@@ -0,0 +1,180 @@
+== Insufficient Resource Allocation
+
+When you attempt to provision the adapter, your adapter might fail.
+
+=== Cause
+
+If your adapter pod does not have sufficient CPU space allocated, your adapter fails as a result.
+
+=== Diagnose
+
+To diagnose the issue: 
+
+. Check the adapter status by running the following command:
++
+`$ ./service-mesh adapter list`
++
+The status is displayed in a few minutes.
+. If the adapter displays a Failed status, run the following command:
+`$ kubectl -n service-mesh get pods`
+. Check the status of the adapter pod: `grpmule-<namespace>`. 
++
+If the status is in Pending state, describe the pod:
++
+`$ kubectl -n service-mesh describe pods -l namespace=<namespace>`.
+. If you do not view the `grpmule-<namespace>` pod, run the same command right after creating the adapter, while it is being provisioned but before it has failed:
++
+`$ kubectl -n service-mesh describe pods -l namespace=<namespace>`.
+. In the Events section, verify that you have an “Insufficient CPU” or other resource listing:
+//need a text of the screenshot here
+. Alternatively, you can describe the nodes to check the resource allocation:
++
+`$ kubectl describe nodes`.
+The allocated resources are displayed:
+//need a text of the screenshot here
+
+=== Solution
+
+To resolve this issue:
+
+. Delete the exisiting adapter.
+. Assign additional resources to your Kubernetes Cluster, as required.
+. Provision your adapter again.
+
+[[image-cannot-be-pulled]]
+== Image Cannot be Pulled
+
+After provisioning the adapter, it fails to start with the following error:
+
+//provide the exact error here
+
+=== Cause
+
+This issue can occur due to invalid credentials specified in the adapter or can occur due to issues with the registry-credentials-rotator pod.
+
+== Diagnosis
+
+To diagnose the issue, verify the status of the affected pods: 
++
+`$ kubectl -n service-mesh get pods`.
+Your pod might appear in either a `ErrImagePull` or a `ImagePullBackOff` state.
+
+=== Solution
+
+To resolve this issue:
+
+. Ensure that the registry-credentials-rotator pod is running:
++
+`$ kubectl -n service-mesh get pod -l app=service-mesh-registry-credentials-rotator`.
+. Review the logs of the pod to determine if any incorrect client ID and client secret, no entitlement, or permissions issues occurred:
++
+`$ kubectl -n service-mesh logs`
++
+`$(kubectl -n service-mesh get pod -l app=service-mesh-registry-credentials-rotator -oname)`.
+. Verify that the pull secret was created:
++
+`$ kubectl -n service-mesh get secret service-mesh-pull-secret`.
+. Verify that the correct platform URL and credentials are provided:
++
+`$ kubectl -n service-mesh get secret service-mesh-registry-credentials-rotator-env -ogo-template='CLIENT_ID:{{.data.CLIENT_ID|base64decode}} CLIENT_SECRET:{{.data.CLIENT_SECRET|base64decode}}'`.
++
+`$ kubectl -n service-mesh get cm service-mesh-registry-credentials-rotator-env  -ogo-template='PLATFORM_URI:{{.data.PLATFORM_URI}}`.
```

Comment:
Please check the spacing in this syntax.

Reply:
```suggestion
`$ kubectl -n service-mesh get cm service-mesh-registry-credentials-rotator-env  -ogo-template='PLATFORM_URI:{{.data.PLATFORM_URI}}`
```

Reply:
Done! Thanks!